learning-another-language/semicolons.md

```diff
@@ -0,0 +1,117 @@
+# Semicolons
+
+## Introduction: To Use or Not Use
+
+In JavaScript, semicolons are used to mark the end of a statement:
+
+```js
+console.log('Do I need to put a semicolon at the end of this to work?');
+```
+
+Officially, JavaScript expects semicolons to end the following kinds of statements:
+
+- variable assignment
+- expressions (such as `1 + 2;`)
+- do-while blocks
+- `continue` statements
+- `break` statements
+- `return` statements
+- `throw` statements to throw errors
+
+Additionally, JavaScript expects semicolons at the end of variable declarations (covered later), `import`s, and `export`s.
+
+However, if we take out the semicolon of our original code, our JavaScript code runs just fine!
+
+```js
+console.log('Do I need to put a semicolon at the end of this to work?')
+```
+
+Semicolons are optional. If semicolons are excluded, then JavaScript's Automatic Semicolon Insertion will... automatically insert semicolons into our code.
+
+### !callout-info
+
+## Developer Conversations in Real-Time
+
+This is one of many topics that the developer community talks about all the time!
+
+### !end-callout
+
+## JavaScript's Automatic Semicolon Insertion
+
+Automatic Semicolon Insertion (ASI) is a subprogram in JavaScript. It executes at the beginning of running our code, and it inserts any missing semicolons.
```

Comment:
subsystem aligns more with how I think of ASI. And from my reading, it sounds more like it happens right along with the program parsing, as opposed to something like a preprocessor phase. 
```suggestion
Automatic Semicolon Insertion (ASI) is a subsystem in JavaScript. It executes as the JavaScript runtime inspects our code, and it attempts to insert any missing semicolons.
```

Reply:
Makes sense/aligns with my understanding too!